Objective-C/Tests/DatabaseTest.m

```diff
@@ -952,6 +952,240 @@ - (void) testPurgeDocOnDeletedDB {
     }];
 }
 
+- (void) testPurgeDocumentOnADeletedDocument {
+    CBLDocument* document = [self generateDocumentWithID: nil];
+    NSString* documentID = document.id;
+    AssertNotNil(documentID);
+    
+    // Delete doc
+    NSError* errorWhileDeletion;
+    Assert([self.db deleteDocument: document error: &errorWhileDeletion]);
+    AssertNil(errorWhileDeletion);
+    
+    // Purge doc on the deleted document
+    NSError* errorWhilePurging;
+    Assert([self.db purgeDocument: document error: &errorWhilePurging]);
+    AssertNil(errorWhilePurging);
+    AssertEqual(self.db.count, 0u);
```

Comment:
is there another way to test whether the document is completely purged from the db?

Reply:
You can do if this one return nil or not: 
```
[[CBLDocument alloc] initWithDatabase: self
                                          documentID: documentID
                                      includeDeleted: YES
                                               error: outError];
```